### ISSUE TITLE
Regression in autodoc

### ISSUE THREAD

**Issue by danielballan** [sphinx-doc/sphinx#6452]
**Describe the bug**
A regression from 2.0.1 to 2.1.0 breaks some aspect of autodoc. I don't know enough about sphinx internals to say exactly what, but I have provided copy/pasteable steps to reproduce below.

**To Reproduce**

Setup:
```
python -mvenv /tmp/sphinx-autodoc-regression
source /tmp/sphinx-autodoc-regression/bin/activate
git clone https://github.com/bluesky/databroker
cd databroker/
pip install wheel
pip install .
pip install -r test-requirements.txt
pip install -r docs-requirements.txt 
```

Reproduce success on 2.0.1:
```
sphinx-build --version  # should be 2.0.1
make -C doc html  # runs successfully
```

Reproduce failure on 2.1.0:

```
pip install sphinx==2.1.0
make -C doc clean
make -C doc html
```

**Expected behavior**
I expect it to succeed on 2.1.0 same as it does on 2.0.1.

**Your project**
https://github.com/bluesky/databroker

**Environment info**
- OS: Ubuntu Linux
- Python version: 3.6.7
- Sphinx version: 2.1.0
- Sphinx extensions:  sphinx.ext.autodoc

**Comment by tk0miya** [comment]
Note: I found an error log.

>don't know which module to import for autodocumenting 'event_sources_by_name' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
https://travis-ci.org/bluesky/databroker/jobs/541963561#L1436

**Comment by tk0miya** [comment]
@danielballan On my local, it failed with following error. Is your procedure right?
```
...
reading sources... [ 98%] tutorial

Exception occurred:
  File "/usr/local/lib/python3.7/site-packages/IPython/sphinxext/ipython_directive.py", line 567, in process_input
    raise RuntimeError('Non Expected exception in `{}` line {}'.format(filename, lineno))
RuntimeError: Non Expected exception in `/databroker/doc/source/tutorial.rst` line None
The full traceback has been saved in /tmp/sphinx-err-nb_4l9no.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!


>>>-------------------------------------------------------------------------
Exception in /databroker/doc/source/tutorial.rst at block ending on line None
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-1-1676ba6f60f9> in <module>
----> 1 with open(path, 'w') as f:
      2     yaml.dump(config, f)

FileNotFoundError: [Errno 2] No such file or directory: '/root/.config/databroker/example.yml'
<<<-------------------------------------------------------------------------
```

**Comment by tk0miya** [comment]
Finally, I found autosummary crashes when creating document for properties.
I'll fix it soon.
Thanks,

**Comment by tk0miya** [comment]
Fixed by #6452.
Thank you for reporting.

**Comment by danielballan** [comment]
Thanks @tk0miya! Apologies for being too slow to respond to your initial questions and for omitting the error log from my initial report. Very much appreciate your work on sphinx!
